symbolic-il2cpp/src/usymlite.rs

```diff
@@ -6,13 +6,116 @@
+    BadOperatingSystem,
+    /// The architecture is missing or can't be read.
+    BadArchitecture,
+    /// A part of the file is not encoded in valid UTF-8.
+    BadEncoding,
+}
+
+impl fmt::Display for UsymLiteErrorKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            UsymLiteErrorKind::MisalignedBuffer => write!(f, "misaligned pointer to buffer"),
+            UsymLiteErrorKind::BadHeader => write!(f, "missing or undersized header"),
+            UsymLiteErrorKind::BadMagic => write!(f, "missing breakpad symbol header"),
+            UsymLiteErrorKind::InvalidVersion => write!(f, "invalid version number"),
+            UsymLiteErrorKind::BadLineCount => write!(f, "unreadable record count"),
+            UsymLiteErrorKind::BufferSmallerThanAdvertised => {
+                write!(f, "buffer does not contain all data header claims it has")
+            }
+            UsymLiteErrorKind::MissingStringTable => write!(f, "string table is missing"),
+            UsymLiteErrorKind::UnterminatedStringTable => {
+                write!(f, "string table does not end with a NULL byte")
+            }
+            UsymLiteErrorKind::BadLines => {
+                write!(f, "could not construct list of source records")
+            }
+            UsymLiteErrorKind::BadId => write!(f, "assembly ID is missing or unreadable"),
+            UsymLiteErrorKind::BadName => write!(f, "assembly name is missing or unreadable"),
+            UsymLiteErrorKind::BadOperatingSystem => {
+                write!(f, "operating system is missing or unreadable")
+            }
+            UsymLiteErrorKind::BadArchitecture => {
+                write!(f, "architecture is missing or unreadable")
+            }
+            UsymLiteErrorKind::BadEncoding => {
+                write!(f, "part of the file is not encoded in valid UTF-8")
+            }
+        }
+    }
+}
+
+/// An error when dealing with [`BreakpadObject`](struct.BreakpadObject.html).
+#[derive(Debug, Error)]
+#[error("{kind}")]
+pub struct UsymLiteError {
+    kind: UsymLiteErrorKind,
+    #[source]
+    source: Option<Box<dyn Error + Send + Sync + 'static>>,
+}
+
+impl UsymLiteError {
+    /// Creates a new Breakpad error from a known kind of error as well as an arbitrary error
+    /// payload.
+    fn new<E>(kind: UsymLiteErrorKind, source: E) -> Self
+    where
+        E: Into<Box<dyn Error + Send + Sync>>,
+    {
+        let source = Some(source.into());
+        Self { kind, source }
+    }
+
+    /// Returns the corresponding [`UsymLiteErrorKind`] for this error.
+    pub fn kind(&self) -> UsymLiteErrorKind {
+        self.kind
+    }
+}
+
+impl From<UsymLiteErrorKind> for UsymLiteError {
+    fn from(kind: UsymLiteErrorKind) -> Self {
+        Self { kind, source: None }
+    }
+}
+
+// TODO: Follow the same structure as usyms and introduce a raw module and other
+// types to distinguish between raw and parsed reps?
```

Comment:
probably, though in general don't sink too much time into this format as we won't use this at all.  It's more to have some reference of what it looks like.

Reply:
gotcha, i'm going to mostly ignore this format for the time being then